docs: document ES|QL data source and fields browsers in Discover (#5701) (#6082)
elastic/docs-content: explore-analyze/discover/try-esql.md

3. Click **Save** to save the query and visualization to a dashboard.

## Browse indices and fields from the editor [discover-esql-resource-browsers]
```{applies_to}
stack: ga 9.4
serverless: ga
```

> When you write a query against your own data, the {{esql}} editor includes two interactive browsers that help you find data sources and field names without leaving the editor:

Author: ```suggestion
When you write a query, the {{esql}} editor includes two interactive browsers that help you find available data sources and field names:
```

> - **Data source browser**: lists the data sources of the following types that you can query: **Alias**, **Index**, **Integration**, **Lookup Index**, **Stream**, and **Timeseries**. Selections are inserted into the `FROM` or `TS` command and existing sources stay preserved. When the query starts with `TS`, only time series data sources are listed.

Reviewer: there's a nascent external **data sources** project afoot in ES|QL which might result in a confusing naming collision here

wonder if we could use different terminologies here, something like this maybe:

- **Index type browser**: lists the indices and data stream types available to query: 

Otherwise, "index abstractions" is the internal umbrella term we're using for these

Author: Yeah we've been chatting about this in some internal threads with @stratoula and some more folks. We didn't land on any decision but I think we at least had a common opinion that data source is the right name there, it can be external at some point, sure.

It doesn't make much sense to try to use some synonyms that will end up confusing users more than having clear terminology with well defined nouns and clear qualifiers/adjectives for identifying the differences.

Kind of what we failed to do with serverless projects and hosted deployments and to this day we're still struggling with it :D

Reviewer: I don't disagree 👍

> - **Data source browser**: lists the data sources you can query, grouped into the following types: **Alias**, **Index**, **Integration**, **Lookup Index**, **Stream**, and **Timeseries**. Selections are inserted into the `FROM` or `TS` command and existing sources stay preserved. When the query starts with `TS`, only time series data sources are listed.

Author: ```suggestion
- **Data source browser**: lists the data sources of the following types that you can query: **Alias**, **Index**, **Integration**, **Lookup Index**, **Stream**, and **Timeseries**. Selections are inserted into the `FROM` or `TS` command and existing sources stay preserved. When the query starts with `TS`, only time series data sources are listed.
```
